blockchain/vm/interpreter.go

```diff
@@ -123,19 +123,18 @@ func NewEVMInterpreter(evm *EVM) *EVMInterpreter {
 		cfg.JumpTable = jt
 	}
 
-	// Enable the opcode computation cost limit
+	// Set the opcode computation cost limit by the default value
 	if cfg.ComputationCostLimit == 0 {
 		switch {
 		case evm.chainRules.IsCancun:
 			cfg.ComputationCostLimit = uint64(params.OpcodeComputationCostLimitCancun)
 		default:
 			cfg.ComputationCostLimit = uint64(params.OpcodeComputationCostLimit)
 		}
-	}
-
-	// It is an experimental feature.
-	if params.OpcodeComputationCostLimitOverride != 0 {
-		cfg.ComputationCostLimit = params.OpcodeComputationCostLimitOverride
+		// It is an experimental feature.
+		if params.OpcodeComputationCostLimitOverride != 0 {
+			cfg.ComputationCostLimit = params.OpcodeComputationCostLimitOverride
+		}
```

Comment:
Could you clarify why you moved this override to the inner scope? While it may not pose a problem, placing it outside appears to be more reliable.

Reply:
The purpose of this change is not to override the computation cost limit of api calls even override value exists. The previous code limits the computation cost of api calls when override value is set.

Reply:
For that purpose, I think that putting additional condition seems appropriate. (e.g., `if cfg.ComputationCostLimit != params.OpcodeComputationCostLimitInfinite`). The changed code has an implicit side effect which may degrade readability.

Reply:
@hyunsooda What about this? This code implies that the computationCostLimit should be set either default values or Infinite.
```
	// Already computation cost limit exists. Infinite value should be only set with api calls.
	if cfg.ComputationCostLimit == params.OpcodeComputationCostLimitInfinite {
		return &EVMInterpreter{evm: evm, cfg: cfg}
	}

	// Set the computation cost with experiment value.
	if params.OpcodeComputationCostLimitOverride != 0 {
		cfg.ComputationCostLimit = params.OpcodeComputationCostLimitOverride
		return &EVMInterpreter{evm: evm, cfg: cfg}
	}

	// Set the opcode computation cost limit by the default value
	switch {
	case evm.chainRules.IsCancun:
		cfg.ComputationCostLimit = uint64(params.OpcodeComputationCostLimitCancun)
	default:
		cfg.ComputationCostLimit = uint64(params.OpcodeComputationCostLimit)
	}
	return &EVMInterpreter{evm: evm, cfg: cfg}
```
The computation cost limit can not be zero, then I think it's good to change the next code too.
```
		}
		// We limit tx's execution time using the sum of computation cost of opcodes.
		in.evm.opcodeComputationCostSum += operation.computationCost
		if in.evm.opcodeComputationCostSum > in.evm.Config.ComputationCostLimit {
			return nil, ErrOpcodeComputationCostLimitReached
		}
		var memorySize uint64
```

Reply:
Looks better. A comment may be attached that highlights the options' priority.

Reply:
Does the priority go like this?
```
1. Input cfg.ComputationCostLimit   (e.g. In klay_call API handler)
2. params.OpcodeComputationCostLimitOverride   (Global override via --opcode-computation-cost-limit)
3. params.OpcodeComputationCostLimit* depending on fork rules   (block sync)
```

Reply:
@blukat29's comment looks clear to me. Please review it again before merging.